### PR TITLE
opencv4: 4.7.0 -> 4.9.0

### PR DESCRIPTION
**Pull request by mweinelt**
https://github.com/opencv/opencv/wiki/ChangeLog#version490

## Description of changes

- Supprt for Python 3.12
- GStreamer support is now effectively limited by availability of elfutils, so no longer on darwin
- VAAAPI support is now available on !darwin

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/opencv/4.x.nix [hunk 1]
```diff
@@ -315,11 +307,12 @@ effectiveStdenv.mkDerivation {
     ++ lib.optional enableFfmpeg ffmpeg
     ++ lib.optionals (enableFfmpeg && effectiveStdenv.isDarwin)
       [ VideoDecodeAcceleration bzip2 ]
-    ++ lib.optionals enableGStreamer (with gst_all_1; [ gstreamer gst-plugins-base gst-plugins-good ])
+    ++ lib.optionals (enableGStreamer && effectiveStdenv.isLinux) (with gst_all_1; [ gstreamer gst-plugins-base gst-plugins-good elfutils orc libunwind zstd ])
     ++ lib.optional enableOvis ogre
     ++ lib.optional enableGPhoto2 libgphoto2
     ++ lib.optional enableDC1394 libdc1394
     ++ lib.optional enableEigen eigen
+    ++ lib.optional enableVA libva
```
Comment (SomeoneSerge) [inline pkgs/development/libraries/opencv/4.x.nix:315]: Nit: I feel like changing from `optional` to `optionals` whenever a line is to be touched anyway

File: pkgs/development/libraries/opencv/4.x.nix
Comment (SomeoneSerge) [inline pkgs/development/libraries/opencv/4.x.nix:1]:
```

       >   Reason: tried: '/usr/local/lib/libopencv_calib3d.409.dylib' (no such file), '/usr/lib/libopencv_calib3d.409.dylib' (no such file, not in dyld cache)
```

I guess this only happens on darwin, shall we just ignore this

**Review by SomeoneSerge — APPROVED** [review]

File: pkgs/development/libraries/opencv/4.x.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/development/libraries/opencv/4.x.nix:315]: I did a thing.

Comment (SomeoneSerge) [reply pkgs/development/libraries/opencv/4.x.nix]: The radical approach. I'm happy

File: pkgs/development/libraries/opencv/4.x.nix
Comment (mweinelt) [reply pkgs/development/libraries/opencv/4.x.nix:1]: Not sure, is this a build time issue?

Comment (SomeoneSerge) [reply pkgs/development/libraries/opencv/4.x.nix:1]: Build time, the passthru tests, aarch64-darwin, as attempted by ofborg in the previous iteration

Comment (mweinelt) [reply pkgs/development/libraries/opencv/4.x.nix:1]: Fails the same on master with 4.7.0. :shrug:

**Pull request merged**